docs/proposals/buildstrategy-steps-resources.md

```diff
@@ -0,0 +1,152 @@
+  define particular resources for each step.
+
+### Non-Goals
+
+- Defining the concrete numbers for the resources(_e.g. CPU/memory_) used per steps in the different Build/ClusterBuildStrategy strategies.
+
+## Proposal
+
+Currently we assign the numbers inside the Build `spec.resources` into all the `steps` for the related strategy. We want to modify this
+by only applying those resources to a single step(_container_). The mechanism for doing this will consist of three parts:
+
+1. We will introduce a new field under the `BuildStep` struct, see `buildstrategy_types.go`. This field will be of the type `boolean`. We propose the following:
+   ```go
+    // BuildStep defines a partial step that needs to run in container for
+    // building the image.
+    type BuildStep struct {
+    	ResourceEnabled  bool `yaml:"resourceEnable,omitempty"`
+    	corev1.Container `yaml:",inline"`
+    }
+   ```
+2. The Build/ClusterBuildStrategy admin will be able to set the above field via a key on the step he/she considers is the more resource consumer.
+   ```yaml
+    - name: step-build
+      image: $(build.builder.image)
+      resourceEnable: true
+      securityContext:
+        runAsUser: 1000
+      command:
+        - /cnb/lifecycle/builder
+      args:
+        - -app=/workspace/source/$(build.source.contextDir)
+        - -layers=/layers
+        - -group=/layers/group.toml
+        - -plan=/layers/plan.toml
+      volumeMounts:
+        - name: layers-dir
+          mountPath: /layers
+   ```
+3. We need to use the above flag in the logic, inside the `generate_taskrun_test.go`, during the loop into all of the strategy steps.
+
+### Risks and Mitigations
+
+For the end-user the only concern is that only one container resources could be modified per strategy. This should be the container in the strategy
+consuming the highest number of resources. Build/ClusterBuildStrategy admins can always modify the steps manually, modifying any step with the resources they want.
+
+## Design Details
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+#### Examples
+
+##### Dev Preview -> Tech Preview
+
+- N/A
+
+##### Tech Preview -> GA
+
+- N/A
+
+**For non-optional features moving to GA, the graduation criteria must include end to end tests.**
+
+##### Removing a deprecated feature
+
+- N/A
+
+### Upgrade / Downgrade Strategy
+
+- N/A
+
+### Version Skew Strategy
+
+- N/A
+
+## Implementation History
+
+- N/A
+
+## Drawbacks
+
+End-Users are limited to only be able to define a single set of values for resources in N strategy step.
+If the end-user would required to set different values for different strategy steps, he/she will not be able to do this.
+However this implies that the user have good knowledge on the used Build/ClusterBuildStrategy strategy, which is not
+supposed to be the case.
+
```

Comment:
I would mention here that the resources set in the build or buildrun still do not reflect the overall resources that the buildrun will consume because there are still other steps beside the one (or those) with the flag set to `true`.

Reply:
yes we can do this,but we might be fixing this now.